controllers/release/adapter_test.go

```diff
@@ -322,6 +330,192 @@ var _ = Describe("Release adapter", Ordered, func() {
 		})
 	})
 
+	When("EnsureFinalPipelineIsProcessed is called", func() {
+		var adapter *adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+			adapter.releaseServiceConfig = releaseServiceConfig
+		})
+
+		It("should do nothing if the Release final pipeline is complete", func() {
+			adapter.release.MarkFinalPipelineProcessing()
+			adapter.release.MarkFinalPipelineProcessed()
+
+			result, err := adapter.EnsureFinalPipelineIsProcessed()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.release.IsFinalPipelineProcessing()).To(BeFalse())
+		})
+
+		It("should do nothing if the Release managed pipeline processing has not yet completed", func() {
+			adapter.release.MarkManagedPipelineProcessing()
+
+			result, err := adapter.EnsureFinalPipelineIsProcessed()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.release.IsFinalPipelineProcessing()).To(BeFalse())
+		})
+
+		It("should do nothing if the Release tenant pipeline processing has not yet completed", func() {
```

Comment:
The tenant pipeline status has no bearing on the final pipeline being processed

Reply:
I have removed it